src/Cooked/Skeleton.hs

```diff
@@ -358,7 +357,14 @@ data TxOpts = TxOpts
     -- | How to resolve anchor in proposal procedures
     --
     -- Default is 'AnchorResolutionLocal Map.Empty'
-    txOptAnchorResolution :: AnchorResolution
+    txOptAnchorResolution :: AnchorResolution,
+    -- | Whether to automatically fill up reference scripts in redeemers when
+    -- possible. This will imply going through all the known utxos with
+    -- reference scripts and compare their hashes, thus will slightly reduce
+    -- performance, especially when handling a lot of utxos.
+    --
+    -- Defaut is 'True'.
```

Comment:
From our current test suite, the overhead is minimal, also it takes a slightly bit more time. I have thought about such optimization, but it has drawbacks:
* it duplicates information
* This map would be hard to maintain because utxos disappear when consumed. So whenever a transaction is validated, we would have to check whether utxos from this map have been consumed to remove them, but only do that in case of successful validation of the transaction.
Overall I believe the current solution is better (because simpler) unless we discover at some point that is really hurts performance.
Also, if we want to address possible future performance issues, we know that this is an area that can offer some improvement.

Reply:
All of this is debatable. This optimization is clearly doable in practice.

Reply:
I think this topic deserves to be mentioned. The alternative implementation should be described in a comment and the two sides of the tradeoff listed.

In my opinion, duplicated information is not a problem. The overhead in terms of space is minimal (there will never be more than a handful of entries in that map and even if we had millions, what is a few MB of ram). However we already often find execution time to be a bit long. Not on one single trace, but in a test suite / audit test suite it quickly adds up to annoying seconds.

On the one hand, space is not really an issue, and there are never a lot of ref scripts anyways so the map will always be small. On the other hand, having a lot of utxos is not uncommon at all: big initial distributions, or sometimes we even do stress tests with a huge number of utxos during certain specific attack attempts.

Therefore I think in terms of scaling, the alternative implementation with a map of known ref scripts and their location is way better.

I would be in favor of either:
- switch to the second implementation, even if it involves a few additions during validation to update the map
- OR keep the current implementation for the time being but turn it off by default and explain the tradeoff in the documentation